trustyai-explainability/trustyai-service-operator
# Enable job suspend for Kueue

Enable job suspend for [Kueue integration](https://kueue.sigs.k8s.io/docs/tasks/dev/integrate_a_custom_job/#overview-of-requirements); however this feature should work independent of Kueue as well.
How to test this feature:
```
ENABLED_SERVICES=LMES CONTROLLER_TOOLS_VERSION=v0.16.3 IMG=quay.io/tedchang/trustyai-service-operator:latest make run
```
Create a job in suspend state. Verify job is in suspend state job's pod is not running.
```
cat << EOF| kubectl apply -f -
apiVersion: trustyai.opendatahub.io/v1alpha1
kind: LMEvalJob
metadata:
  labels:
    app.kubernetes.io/name: fms-lm-eval-service
    app.kubernetes.io/managed-by: kustomize
  name: evaljob-sample
  namespace: default
spec:
  suspend: true
  model: hf
  modelArgs:
  - name: pretrained
    value: EleutherAI/pythia-70m
  taskList:
    taskNames:
    - unfair_tos
  logSamples: true
  limit: "5"
EOF
```

Set suspend to false and verify job's pod getting created and running
```
oc patch lmevaljob evaljob-sample --patch '{"spec":{"suspend":false}}' --type merge
```
